thirdeye-ui/src/app/pages/alerts-anomalies-page/alerts-anomalies-page.component.tsx

```diff
@@ -90,13 +114,29 @@ export const AlertsAnomaliesPage: FunctionComponent = () => {
             startTime,
             endTime,
             enumerationItemId: Number(enumerationItemIdStr),
-        }).then((anomalies) => {
-            if (anomalies) {
-                setUiAnomalies(getUiAnomalies(anomalies));
-            }
+            showAllAnomalies: true,
         });
     };
 
+    const getFilteredUiAnomalies = useCallback(
+        (anomaliesProp: typeof anomalies): UiAnomaly[] | null => {
+            if (!anomaliesProp) {
+                return null;
+            }
+
+            // All anomalies being fetched, and then filtered here at UI level
+            const filteredAnomalies = filterAnomaliesByFunctions(
+                anomaliesProp,
+                showAllAnomalies ? [] : [(a) => !isAnomalyIgnored(a)]
+            );
+
+            return getUiAnomalies(filteredAnomalies);
+        },
+        [showAllAnomalies]
+    );
+
+    const uiAnomalies = getFilteredUiAnomalies(anomalies);
```

Comment:
Shouldn't `uiAnomalies` just be memoized depending on `showAllAnomalies` and `anomalies`?

Reply:
That's a good idea. Simplified this by implementing just a `useMemo`